src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -239,6 +240,20 @@ module.exports = function<T, P, I, TI, C, CX>(
             currentHostContext,
             workInProgress
           );
+
+          // A component cannot be focused until it's mounted.
+          // Flag components with autoFocus for later work.
+          switch (type) {
+            case 'button':
+            case 'input':
+            case 'select':
+            case 'textarea':
```

Comment:
These types are specific to DOM renderer so they shouldn’t be in the reconciler.